src/service/resources-impl.js

```diff
@@ -1009,16 +1009,48 @@ export class ResourcesImpl {
         ) {
           // 7. The new height (or one of the margins) is smaller than the
           // current one.
-        } else if (
-          request.newHeight == box.height &&
-          this.isWidthOnlyReflowFreeExpansion_(
-            resource.element,
-            request.newWidth || 0
-          )
-        ) {
-          // 8. Element is in viewport, but this is a width-only expansion that
-          // should not cause reflow.
-          resize = true;
+        } else if (request.newHeight == box.height) {
+          // 8. Element is in viewport, but this is a width-only expansion.
+          // Check whether this should be reflow-free, in which case,
+          // schedule a size change.
+          this.vsync_.run(
+            {
+              measure: state => {
+                const parent = resource.element.parentElement;
+                if (!parent) {
+                  state.resize = false;
+                  return;
+                }
+
+                // If the element has siblings, it's possible that a width-expansion will
+                // cause some of them to be pushed down.
+                const parentWidth =
+                  (parent.getLayoutWidth && parent.getLayoutWidth()) ||
+                  parent./*OK*/ getBoundingClientRect().width;
+                let cumulativeWidth = widthDiff;
+                for (let i = 0; i < parent.childElementCount; i++) {
```

Comment:
What if this is a structure like this:

```
                   super-parent
                  /            \
    kind-of-useless-parent     actual-sibling
            |
    resizing-element
```

Reply:
I don't think this should be an issue. <resizing-element> will never grow wider than <kind-of-useless-parent>, so the invariant that width(<kind-of-useless-parent>) + width(<actual-sibling>) < width(viewport) should hold.